src/utils/utils_load_library.h

```diff
@@ -39,22 +39,34 @@ static inline void *util_open_library(const char *filename) {
 static inline int util_close_library(void *handle) {
     // If the FreeLibrary function succeeds, the return value is nonzero.
     // If the FreeLibrary function fails, the return value is zero.
-    return (FreeLibrary(handle) == 0);
+    return (FreeLibrary((HMODULE)handle) == 0);
 }
 
-static inline void *util_get_symbol_addr(void *handle, const char *symbol) {
-    return GetProcAddress(handle, symbol);
+static inline void *util_get_symbol_addr(void *handle, const char *symbol,
+                                         const char *libname) {
+    if (handle == NULL) {
+        if (libname == NULL) {
+            return NULL;
+        }
+        handle = GetModuleHandle(libname);
+    }
+    return GetProcAddress((HMODULE)handle, symbol);
 }
 
 #else /* Linux */
 
 static inline void *util_open_library(const char *filename) {
-    return dlopen(filename, RTLD_LAZY);
+    // NOTE that we use RTLD_GLOBAL which add all loaded symbols to the
+    // global symbol table. These symbols would be used by the Level Zero
+    // provider later
+    return dlopen(filename, RTLD_GLOBAL | RTLD_LAZY);
 }
 
 static inline int util_close_library(void *handle) { return dlclose(handle); }
 
-static inline void *util_get_symbol_addr(void *handle, const char *symbol) {
+static inline void *util_get_symbol_addr(void *handle, const char *symbol,
+                                         const char *libname) {
+    (void)libname; //unused
     return dlsym(handle, symbol);
```

Comment:
```suggestion
    if (!handle) 
        handle = RTLD_DEFAULT;
    return dlsym(handle, symbol);
```
Technically it doesn't matter - as RTLD_DEFAULT is NULL - but it will increase readability of the code, and it will make it more similar to windows one

Reply:
Okey

Reply:
Could you take a look at this.? I have no such symbol in dlfcn.h. Are you sure is it essential for readability of the code?

Reply:
https://github.com/oneapi-src/unified-memory-framework/actions/runs/8836639498/job/24263681354?pr=435

Reply:
#define _GNU_SOURCE 1

>        The _GNU_SOURCE feature test macro must be defined in order to obtain the definitions  of  RTLD_DEFAULT  and
>        RTLD_NEXT from <dlfcn.h>.
>

Reply:
it still doesn't work, I manually added the RTDL_DEFAULT declarations

Reply:
I fix that. I need to declare it in other places

Reply:
i know why this issue happens - other .c file include header before including this one - correct solution whould be to move implementation of this utils function to .c file - but i'm not sure how much work this requires. 

if it is alot of work it whould be better to just change order of includes in this files where it is required(, instead of defining GNU_SOURCE in those files

Reply:
or better idea - just add this define to cmake - and remove all other gnu_source from source.